src/resolution/resolver.ts

```
@@ -51,103 +51,128 @@ const _resolveRequest = <T>(requestScope: interfaces.RequestScope) =>
+    request: interfaces.Request,
+    binding:interfaces.Binding<T>) => {
+    let result: T | Promise<T>;
+    const childRequests = request.childRequests;
+    if (binding.type === BindingTypeEnum.ConstantValue) {
+        result = binding.cache!;
+        binding.activated = true;
+    } else if (binding.type === BindingTypeEnum.Function) {
+        result = binding.cache!;
+        binding.activated = true;
+    } else if (binding.type === BindingTypeEnum.Constructor) {
+        result = binding.implementationType as T;
+    } else if (binding.type === BindingTypeEnum.DynamicValue && binding.dynamicValue !== null) {
+        result = invokeFactory(
+            "toDynamicValue",
+            binding.serviceIdentifier,
+            () => (binding.dynamicValue!)(request.parentContext)
+        );
+    } else if (binding.type === BindingTypeEnum.Factory && binding.factory !== null) {
+        result = invokeFactory(
+            "toFactory",
+            binding.serviceIdentifier,
+            () => (binding.factory!)(request.parentContext)
+        );
+    } else if (binding.type === BindingTypeEnum.Provider && binding.provider !== null) {
+        result = invokeFactory(
+            "toProvider",
+            binding.serviceIdentifier,
+            () => (binding.provider!)(request.parentContext)
+        );
+    } else if (binding.type === BindingTypeEnum.Instance && binding.implementationType !== null) {
+        result = resolveInstance<T>(
+            binding,
+            binding.implementationType as interfaces.Newable<T>,
+            childRequests,
+            _resolveRequest<T>(requestScope)
+        );
+    } else {
+        // The user probably created a binding but didn't finish it
+        // e.g. container.bind<T>("Something"); missing BindingToSyntax
+        const serviceIdentifier = getServiceIdentifierAsString(request.serviceIdentifier);
+        throw new Error(`${ERROR_MSGS.INVALID_BINDING_TYPE} ${serviceIdentifier}`);
+    }
```

Maybe we could replace this for a `switch` control structure.

```ts

switch (binding.type) {
    case BindingTypeEnum.DynamicValue:
        result = _getREsolvedDynamicValueBinding(...)
        break;
}

...

function _getREsolvedDynamicValueBinding(...) {
    if (binding.dynamicValue) {
        return invokeFactory(
            "toDynamicValue",
            binding.serviceIdentifier,
            () => (binding.dynamicValue!)(request.parentContext)
        );
    } else {
         return undefined;
    }
}

```

We should throw an error if result is `undefined` after this switch:

```ts
    if (result === undefined) {
        // The user probably created a binding but didn't finish it
        // e.g. container.bind<T>("Something"); missing BindingToSyntax
        const serviceIdentifier = getServiceIdentifierAsString(request.serviceIdentifier);
        throw new Error(`${ERROR_MSGS.INVALID_BINDING_TYPE} ${serviceIdentifier}`);
    }
```

---

I will get there.......

---

Done